### PR TITLE
WIP: Fixes #175

### LINKED ISSUE #175: Using wamp.session.list segfaults the router
Hey,

I just tried to use `wamp.session.list` with an auth role filter, as described [here](https://wamp-proto.org/_static/gen/wamp_latest.html#wamp-session-list).

My call using kraftfahrstrasse: 

```
const [sessionListPromise] = socketConnection.Call('wamp.session.list', [['user']]);
```

Router Log:

```
interconnect_1           | panic: interface conversion: interface {} is []interface {}, not []string
interconnect_1           | 
interconnect_1           | goroutine 8 [running]:
interconnect_1           | github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router.(*realm).sessionList(0xc4201989c0, 0xc420179a00, 0x196520c5, 0x8a7d5c35da7c3d88)
interconnect_1           | 	/go/src/github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router/realm.go:829 +0x363
interconnect_1           | github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router.(*realm).(github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router.sessionList)-fm(0xc420179a00, 0xc420021350, 0x2)
interconnect_1           | 	/go/src/github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router/realm.go:267 +0x34
interconnect_1           | github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router.(*realm).metaProcedureHandler(0xc4201989c0)
interconnect_1           | 	/go/src/github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router/realm.go:774 +0x346
interconnect_1           | created by github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router.(*realm).run
interconnect_1           | 	/go/src/github.com/EmbeddedEnterprises/autobahnkreuz/vendor/github.com/gammazero/nexus/router/realm.go:298 +0x6cb
slimerp-service-deploy_interconnect_1 exited with code 2

```

https://github.com/gammazero/nexus/blob/master/router/realm.go#L829

I tried to figure out the issue in the code, but I was not able to and I think, I use this correctly. No matter what, this should not crash the entire router, even if this is not used correctly.

Best Regards, 
  Johann

### PR DESCRIPTION
**Pull request by johannwagner**
https://github.com/gammazero/nexus/issues/175

### CODE REVIEW

File: router/realm.go [hunk 1]
```diff
@@ -789,7 +789,26 @@ func (r *realm) metaProcedureHandler() {
 func (r *realm) sessionCount(msg *wamp.Invocation) wamp.Message {
 	var filter []string
 	if len(msg.Arguments) != 0 {
-		filter, _ = msg.Arguments[0].([]string)
+		filterList, isOk := wamp.AsList(msg.Arguments[0])
+
+		if isOk {
```
Comment (gammazero) [inline router/realm.go]: Should this be `if !isOK {`

File: router/realm.go [hunk 2]
```diff
@@ -789,7 +789,26 @@ func (r *realm) metaProcedureHandler() {
...
+			return &wamp.Error{
+				Error:   wamp.ErrInvalidArgument,
+				Request: msg.Request,
+			}
+		}
+
+		for _, x := range filterList {
+			role, ok := x.(string)
+			if ok {
```
Comment (gammazero) [inline router/realm.go]: No need for `if ... else ...`.  Just say:
```
if !ok {
    return &wamp.Error{ ... }
}
filter = append(filter, role)
```

**Comment by gammazero** [comment]
Definitely need unit tests for this.

**Comment by gammazero** [comment]
I fixed a couple things in the PR and added unit tests.  FYI, the `wamp.Error` returned by the meta procedure needed to have `Type: wamp.INVOCATION`  otherwise it would not be handled correctly.

Please review.

**Comment by martin31821** [comment]
LGTM 👍

**Pull request merged**